src/tunserver.cpp

```diff
@@ -1128,19 +1133,30 @@ void c_tunserver::event_loop(int time) {
 						_note("Using CT tunnel to decrypt data for us");
 						auto & ct = * find_tunnel->second;
 						auto tundata = ct.unbox_ab( blob , nonce_used );
-						if (!check_packet_destination_address(dst_hip, tundata)) {
-							_warn("crypto authentification of destination IP failed");
-							throw std::runtime_error("crypto authentification of destination IP failed");
-						}
-						if (!check_packet_source_address(src_hip, tundata)) {
-							_warn("crypto authentification of source IP failed");
-							throw std::runtime_error("crypto authentification of source IP failed");
-						}
+
 
 						_note("<<<====== TUN INPUT: " << to_debug(tundata));
-                                                auto write_bytes = m_tun_device.write_to_tun(&tundata[0], tundata.size());
+						if (check_ip_protocol(tundata)) {
+							// add TUN header
+							const unsigned char tun_header[] = {0x00, 0x00, 0x86, 0xDD};
+							tundata.insert(g_ipv6_rfc::header_position_of_dst, reinterpret_cast<const char *>(dst_hip.data()), dst_hip.size());
+
+/*							if (!check_packet_destination_address(dst_hip, tundata)) {
```

Comment:
This is commented out because we construct this information above, I suppose it could be left as some (low-probability) assert here.
(idea to discuss)